include/boost/json/impl/value_stack.ipp

```diff
@@ -364,16 +364,25 @@ push_array(std::size_t n)
     st_.exchange(std::move(ua));
 }
 
-void
+error_code
 value_stack::
-push_object(std::size_t n)
+push_object(std::size_t n, bool ignore_duplicates)
 {
     // we already have room if n > 0
     if(BOOST_JSON_UNLIKELY(n == 0))
         st_.maybe_grow();
+
     detail::unchecked_object uo(
-        st_.release(n * 2), n, sp_);
-    st_.exchange(std::move(uo));
+        st_.release(n * 2), n, sp_, ignore_duplicates);
+    st_.exchange(uo);
+
+    error_code ec;
+    // constructed object should have consumed all of uo's data
+    if( uo.size() )
+    {
+        BOOST_JSON_FAIL( ec, error::duplicate_key );
+    }
```

Comment:
I believe that the way this is implemented has impacted performance, which is causing the regression in the per-commit performance checker.